### DISCUSSION TITLE
Hagiwo Sync LFO still WIP?

### DISCUSSION THREAD

**Discussion by awonak** [mzuelch/CATs-Eurosynth#7] [Q&A]
Greetings, 

I was wondering what steps remain keeping the Hagiwo Sync LFO in WIP status? I used the gerbers to order smd populated PCBs & Panels from JLCPCB and everything seems to work for me. I'm able to run all 3 of the of the firmware's listed in this repo. Thanks for sharing these files, this was my first time ordering populated PCBs. I'm shocked how cheap it was!

Cheers,
Adam

**Comment by mzuelch** [comment]
Yes it is, isn't it ;)
My inner monk was a little bit unhappy about its performance and I had planned at the time to improve the firmware.
Over the last months my module was built several times and I never received any complaints about the usability and function - so I think I should kick it to "Working". I actually never bothered mainly because I've focussed on my 2HP Utilities.

**Comment by awonak** [comment]
If you have any specific observations about firmware performance, consider adding issues for them. I'd be happy to help take on an issue as an open source contribution way of saying thanks for sharing the hardware files.

My one hardware gripe is that the arduino nano should be rotated 180 degrees so the usb cable doesn't bump into the module power cable.

I'm also exploring the possibility of creating a new firmware inspired by the [Generative Sequencer](https://note.com/solder_state/n/n8a489f0b857a) firmware for the Sync LFO hardware build. Imagine P1 -> probability of new voltages , P2 -> number of steps, P3 -> voltage amplitude, P4 -> ? (maybe quantized scales or slew rate?). Each cv input would advance the step and possibly generate a new voltage value based on the probability. The bi-polar outputs would add even more sequencing & modulation patching opportunities! 

Building off that idea, since the module layout is symmetrical, I think it would be really neat to have reversible panel designs (similar to Zlob's artistic/soulless panels) with the knob labels for each of the firmware.

Thanks again for sharing all of the files and organizing the firmware in this repo.

Also, Knopsl says Hi from the [EuroPi Discord](https://discord.gg/JaQwtCnBV5)!

**Reply by mzuelch** [reply]
Thanks for the tips - I will look into it as soon as I finished my latest module

**Reply by linxlinus** [reply]
> I'm also exploring the possibility of creating a new firmware inspired by the [Generative Sequencer](https://note.com/solder_state/n/n8a489f0b857a) firmware for the Sync LFO hardware build. Imagine P1 -> probability of new voltages , P2 -> number of steps, P3 -> voltage amplitude, P4 -> ? (maybe quantized scales or slew rate?). Each cv input would advance the step and possibly generate a new voltage value based on the probability. The bi-polar outputs would add even more sequencing & modulation patching opportunities!

That is one great idea. I quite like the Generative Sequencer and have one in my case. Though i like to add a few things like in the original Turing machine. Expanders would be quite nice as well.

I am in for the fun!

**Reply by mzuelch** [reply]
First things first - I've just updated the Sync LFO layout with the Nano rotated 180 deg. There are no changes in pot or jack position so all the panels should work just fine.

@linxlinus/Modulove: You have to keep it in mind when referring your customers to my page for assembly. New Revision Number 022303!

Edit: Just added a revision history as well and sent the Modulove appendix under its own header.

**Reply by linxlinus** [reply]
@mzuelch will make sure the documentation is upto date and links referring the right spot! Thanks for picking this up :) Just love the LFO and tinkering with code ideas again.

**Reply by awonak** [reply]
I spent some time playing around with the generative sequencer idea, and I was able to get a rough and basic working version in less than 100 lines of code! It has some bugs and the PWM isn't as stable as I'd like, so I'm going to explore the FlexiTimer2 library. I'll share my final working version when it's cleaned up.

https://gist.github.com/awonak/517ac0b22aae5088f3950d312ba545af

**Reply by awonak** [reply]
Okay, I'm pretty happy with how this firmware turned out!
https://youtu.be/okj47TcXJXg

**Reply by modulove** [reply]
> Okay, I'm pretty happy with how this firmware turned out!

Very nice! I will give this a try.

**Reply by awonak** [reply]
oh hey! I noticed a space for new coming soon firmware on the [download page](https://dl.modulove.de/module/synclfo/). Mind if I submit a pull request with my firmware? Or do you have something else in mind?

**Reply by earsneyes** [reply]
hah you got me there :). I love the Firmware but couldnt give it a testdrive besides from loading the fw. Please go ahead!

**Reply by earsneyes** [reply]
Was thinking to try it on a cv sequencer module i have here. But work came in the way. 
I have somthing ready for a new board and it runs Hagiwos cv Sequencer Firmware fine. It features an OLED (SPI), Rotary Encoder and two additional CV inputs and i added a few things to the fw. Great fun and inspiration these modules and the work @mzuelch has done is amazing for starters like me!

**Reply by earsneyes** [reply]
> Okay, I'm pretty happy with how this firmware turned out!

I gave it a spin yesterday and i must say, I love it! 
It worked very nice for making rings melodic ands it works as well for a 303 bassline.
Will look into this deeper when time permits :)

**Comment by modulove** [comment]
@mzuelch i suggest to change the position of C11/C8 and D3/D4 so there is space for a shrouded IDC Header.
![image](https://user-images.githubusercontent.com/98668901/227740593-5a665fed-9d23-4ace-929e-6fb9a87feb6a.png)
Done: Should i go ahead and try a pull request ?

**Comment by modulove** [comment]
@mzuelch I suggest to protect the outputs with a diode as well.
Is it possible to add a bicolor LED for the bipolar outputs ?

**Reply by mzuelch** [reply]
A bicolor LED should be doable - all it needs is an opamp, a resistor and the LED itself. But I will have to redesign the lower part of the PCB quite extensively. Just have a look into the schematics of Abakus or Haraldwerk's LFO.
For output protection, a 1206 1k resistor always does the trick. When misusing an output as input, this R limits the current into the last opamp output and prevents the opamp from blowing up.
I would rather prefer to downsize the input protection. The diode network is a little bit too much - I've just copied HAGIWOs design to learn if it's better than my usual design. It isn't.
